README.md

```diff
@@ -60,16 +60,18 @@ tool\
-| unsigned int   | int8, uint16, uint32, uint64 | uint8_t, uint16_t, uint32_t, uint64_t |                                                                 |
-| floating point | fp32, fp64                   | float/double                          |                                                                 |
-| bitfield       | bitfield:<bits>              | uint32_t bf : 3                       | unsigned integer with specified amount of bits                  |
-| string         | string                       | const char*                           | utf8 encoded, null-terminated                                   |
-| inline_array   | int8[20], some_type[20]      | type[20]                              | fixed length, type can be any POD-type or user defined struct   |
-| array          | int8[], some_type[]          | struct { type* data; uint32_t count } | dynamic length, type can be any POD-type or user defined struct |
-| pointer        | some_type*                   | type*                                 | pointer to any user-defined type                                |
+| name           | tld-type                     | C-storage                                   | comment                                                              |
+|----------------|------------------------------|---------------------------------------------|----------------------------------------------------------------------|
+| signed int     | int8, int16, int32, int64    | int8_t, int16_t, int32_t, int64_t           |                                                                      |
+| unsigned int   | int8, uint16, uint32, uint64 | uint8_t, uint16_t, uint32_t, uint64_t       |                                                                      |
+| floating point | fp32, fp64                   | float/double                                |                                                                      |
+| bitfield       | bitfield:<bits>              | uint32_t bf : 3                             | unsigned integer with specified amount of bits                       |
+| string         | string                       | const char*                                 | utf8 encoded, null-terminated                                        |
+| inline_array   | int8[20], some_type[20]      | type[20]                                    | fixed length, type can be any POD-type or user defined struct        |
+| array          | int8[], some_type[]          | struct { type* data; uint32_t count }       | dynamic length, type can be any POD-type or user defined struct      |
+| pointer        | some_type*                   | type*                                       | pointer to any user-defined type                                     |
+| anyptr         | anyptr                       | struct { void* ptr; dl_typeid_t tid; }      | pointer to any struct type, where the type is decided on assignment  |
```

Comment:
for consistency this should probably be 'any_pointer'... and I wished I would have called 'pointer' 'ptr', but the ship has sailed on that one!
same with 'any_array' is consistent with 'inline_array'

Reply:
and should there be an 'any_inline_array' for completeness?

Reply:
Renamed them
any_inline_array needs a known element (max)size, like a union. It would be better to just change the union type to allow explicit max size and assignment of any type to it in that case